containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -77,71 +109,130 @@ def get_account_id() -> str:
     return boto3.client("sts").get_caller_identity()["Account"]
 
 
-class DataProductConfig:
+@dataclass
+class DataProductElement:
     """
-    Configures the name, S3 paths, and athena table names for a data product.
+    A dataset within the data product. The curated data for each element
+    is queryable in its own athena table.
     """
 
-    def __init__(self, name: str, table_name: str, bucket_name: str | None = None):
-        """
-        Generate all the paths based on the data product name and a table name
-        """
-        if bucket_name is None:
-            bucket_name = get_bucket_name()
+    name: str
+    data_product: DataProductConfig
 
-        self.name = name
+    @staticmethod
+    def load(name, data_product_name):
+        data_product = DataProductConfig(name=data_product_name)
+        return DataProductElement(data_product=data_product, name=name)
 
-        self.raw_data_prefix = BucketPath(
-            bucket=bucket_name, key=os.path.join("raw_data", name, table_name) + "/"
+    @property
+    def raw_data_prefix(self):
+        return BucketPath(
+            bucket=self.data_product.raw_data_bucket,
+            key=os.path.join("raw_data", self.data_product.name, self.name) + "/",
         )
 
-        self.curated_data_prefix = BucketPath(
-            bucket_name,
+    @property
+    def curated_data_prefix(self):
+        return BucketPath(
+            self.data_product.curated_data_bucket,
             os.path.join(
                 "curated_data",
-                f"database_name={name}",
-                f"table_name={table_name}",
+                f"database_name={self.data_product.name}",
+                f"table_name={self.name}",
             )
             + "/",
         )
 
-        self.raw_data_table = QueryTable(
-            database=RAW_DATABASE_NAME, name=f"{table_name}_raw"
-        )
-        self.curated_data_table = QueryTable(database=name, name=table_name)
+    def raw_data_table_unique(self):
+        suffix = uuid4()
+        name = f"{self.name}_{suffix}_raw"
+        if len(name) > MAX_IDENTIFIER_LENGTH:
+            raise ValueError(f"Generated table name too long: {name}")
```

Comment:
Do we want to try a shorter UUID if adding a UUID is too long?

The following suggestion is **completely** optional, but [gains 6 characters while losing almost no entropy](https://stackoverflow.com/a/55229089/10478115). It comes at the cost of the random string no longer being visually identifiable as UUID4.

e.g. 
```suggestion
        suffix_lim = suffix.hex[:12] + suffix.hex[13:16] + suffix.hex[17:]
        name = f"{self.name}_{suffix_lim}_raw"
        if len(name) > MAX_IDENTIFIER_LENGTH:
            raise ValueError(f"Generated table name too long: {name}")
```

Reply:
This is a nice suggestion, but I think I'll leave it for now as 255 characters is a lot to work with, and we might end up limiting the name length enough that this is fine anyway. If we hit this edge case in practice then I'm happy to revisit this